packages/nouns-webapp/src/i18n/setLocale.ts

```diff
@@ -0,0 +1,16 @@
+import { dynamicActivate } from './NounsI18nProvider';
+
+/**
+ * Sets locale in local storage
+ *
+ * Note: this value will persist across sessions
+ *
+ * @param locale Locale we wish to use for this user
+ */
+export const setLocale = (locale: string) => {
+  if (localStorage.getItem('lang') !== locale) {
+    localStorage.setItem('lang', locale);
+    window.location.reload();
```

Comment:
What's the purpose of the reload? Shouldn't be required to do a full page refresh on locale change.

Reply:
Could be missing something, but I found that the date localization didn't work until after a reload. Less importantly, this was also to fix issues with a few locale specific UI changes I made (spacing etc.) that look at `useActiveLocale` to trigger.

Reply:
I'll take a look at the date localization issue, but it'd be a much nicer UX if we didn't need to do a full-page refresh.

Reply:
Enabling `forceRenderOnLocaleChange` should solve the same issue and provide a cleaner UX. It will re-render the UI without requiring API data to be re-fetched.

Reply:
I had that enabled (I think) but still required a full reload to get the dates to change ... will also look into this. Fully ack that I could just be being stupid on this...